### ISSUE TITLE
Gradients of the predictions wrt input variables

### ISSUE THREAD

**Issue by cs92** [soda-inria/tabicl#109]
Hi, 

I am new to Tabular foundation models and recently started using TabICL v2 for regression tasks. Its a great library, kudos to the developers for making this open source! The sklearn like interface also makes it super easy to start using it.

For my problem, I am interested in obtaining the sensitivities/gradients of the TabICL predictions with respect to the input variables (e.g., dy/dx1, dy/dx2, ... ). Is this functionality available in TabICL currently? If not, is there a straightforward way to obtain it?

Thanks in advance!

**Comment by dholzmueller** [comment]
Hi, this functionality is not provided directly, and while you could use pytorch autodiff to differentiate the model, you would also need to do the same with the preprocessing, which is currently not torch-based. So this could be a good addition, but would require some refactoring that we currently don't have the capacity for.
